src/protocols/aeternity/composables/aeNames.ts

```diff
@@ -0,0 +1,315 @@
+import {
+  computed,
+  ref,
+  watch,
+} from 'vue';
+import type {
+  ChainName,
+  IDefaultComposableOptions,
+  IName,
+  ITransaction,
+  NetworkId,
+  IAddressNamePair,
+  IAuction,
+  IAuctionBid,
+} from '@/types';
+import {
+  fetchAllPages,
+  fetchJson,
+  handleUnknownError,
+} from '@/utils';
+import { Encoded, NAME_TTL } from '@aeternity/aepp-sdk';
+import {
+  AUTO_EXTEND_NAME_BLOCKS_INTERVAL,
+  PROTOCOL_AETERNITY,
+  STORAGE_KEYS,
+} from '@/constants';
+import { tg } from '@/store/plugins/languages';
+import { useModals } from '@/composables/modals';
+import { useMiddleware } from '@/composables/middleware';
+import { useAccounts } from '@/composables/accounts';
+import { useTransactionList } from '@/composables/transactionList';
+import { useAeSdk } from '@/composables/aeSdk';
+import { createPollingBasedOnMountedComponents, useStorageRef } from '@/composables/composablesHelpers';
+import { isInsufficientBalanceError } from '@/protocols/aeternity/helpers';
+import { UPDATE_POINTER_ACTION } from '@/protocols/aeternity/config';
+import { useAeNetworkSettings } from '@/protocols/aeternity/composables/aeNetworkSettings';
+
+const POLLING_INTERVAL = 10000;
+
+interface IUpdateNamePointerParams {
+  name: ChainName;
+  address?: any;
+  type?: keyof typeof UPDATE_POINTER_ACTION;
+}
+
+interface IAuctionEntryParams {
+  name: ChainName;
+  expiration: number;
+  bids: IAuctionBid[];
+}
+
+type NamesRegistry = Record<NetworkId, Record<Encoded.AccountAddress, ChainName>>;
+
+const ownedNames = useStorageRef<IName[]>([], STORAGE_KEYS.ownedNames);
+const pendingAutoExtendNames = ref<ChainName[]>([]);
+const areNamesFetching = ref(false);
+
+/**
+ * Stores default names for a certain local account addresses in selected network
+ */
+const defaultNamesRegistry = useStorageRef<NamesRegistry>({}, STORAGE_KEYS.defaultNames);
+/**
+ * Stores default names for a certain external account addresses in selected network
+ */
+const externalNamesRegistry = ref<NamesRegistry>({});
+const auctions = ref<Record<string, IAuction>>({});
+
+const initPollingWatcher = createPollingBasedOnMountedComponents(POLLING_INTERVAL);
+
+export function useAeNames({ store }: IDefaultComposableOptions) {
+  const { openDefaultModal } = useModals();
+  const { aeActiveNetworkSettings } = useAeNetworkSettings();
+  const { fetchPendingTransactions } = useTransactionList({ store });
+
+  const {
+    nodeNetworkId,
+    getAeSdk,
+  } = useAeSdk({ store });
+
+  const {
+    isMiddlewareReady,
+    getMiddleware,
+    getMiddlewareRef,
+    fetchFromMiddlewareCamelCased,
+  } = useMiddleware();
+
+  const {
+    aeAccounts,
+    isLocalAccountAddress,
+    getLastActiveProtocolAccount,
+  } = useAccounts();
+
+  function getOwnedName(name: string): IName | undefined {
```

Comment:
If the `getName` function returns string I suggest to rename this method to `getOwnedNameData` or `getOwnedNameDetails` as it returns an object.